src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs

```diff
@@ -207,6 +211,20 @@ protected virtual Task WaitForChangesAsync(RequestContext context, CancellationT
                     }
                 }
 
+                // Now, for any diagnostics reported from a prior source that we do not see this time around, report its
+                // diagnostics as being removed. This allows for different sets of diagnostic-sources to be computed
+                // each time around, while still producing accurate diagnostic reports.
+                //
+                // Only do this if we haven't already created a removal report for that prior result.
+                foreach (var (projectOrDocumentId, previousDiagnosticParams) in documentToPreviousDiagnosticParams)
+                {
+                    if (!seenDiagnosticSourceIds.Contains(projectOrDocumentId) &&
+                        !removedResults.Contains(previousDiagnosticParams))
+                    {
+                        progress.Report(CreateRemovedReport(previousDiagnosticParams.TextDocument));
```

Comment:
Might be good to note here that we're creating a removed report because as far as the client is concerned for diagnostics - the document is removed.

We don't want to create an empty report as that might mean the client thinks we're still reporting diagnostics for it (but with none).

Reply:
sounds good.